src/main/java/com/shapesecurity/functional/data/NonEmptyImmutableList.java

```diff
@@ -22,6 +22,7 @@
 
 import javax.annotation.Nonnull;
 import java.util.ArrayList;
+import java.util.Stack;
```

Comment:
You're importing `java.util.Stack` but not using it. Any reason why you're using arrays instead of `Stack`?

Reply:
The non-trivial overhead of creating `Integer` objects for very large lists is not worth the slight convenience of using `java.util.Stack`.

Reply:
Okay let's remove the import then.

Reply:
Removed